@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -177,9 +218,209 @@ export class MultiPartUploadWrapper {
      * The Promise with the HTTP status or rejects in case of errors.
      */
     async upload(
-        opts: UploadOptions,
+        data: string | File | Blob | ArrayBufferLike | BlobData,
         abotrSignal?: AbortSignal
     ): Promise<number> {
+        /**
+         * The method works in this way:
+         * 1) Starts the MultipartUpload process by sending a request to Blob V1 or V2.
+         * 2) Starts reading the chunk by chunk from data to upload in the main while loop.
+         * 3) After reading the same count of the chunks as the configured maximum parallels upload requests,
+         * the main loop waiting while the read chunks upload to the blob storage in the uploading loop.
+         * 4) After uploading all chunks in the uploading loop, the main loop continues to read and upload
+         * the next chunks of the data.
+         * 5) After uploading all bytes the method submits the MultiPartUpload process by sending a request
+         * with chunk numbers and ids to the Blob V1 or V2 and returns 204 if success
+         * or rejects with an error if any.
+         */
+        const defaultParallelRequests = 6;
+        const parallelRequests =
+            this.opts.parallelRequests || defaultParallelRequests;
+
+        const parts: { id: string; number: number }[] = [];
+
+        const blobData = await this.getBlobData(data);
+
+        const dataSize = blobData.size();
+
+        const request = await this.getUploadRequest(abotrSignal);
+
+        const startMultipartResponse = await request.startMultipartUpload({
+            contentType: this.opts.contentType,
+            handle: this.opts.handle,
+            layerId: this.opts.layerId,
+            billingTag: this.opts.billingTag,
+            contentEncoding: this.opts.contentEncoding
+        });
+
+        if (
+            this.opts.blobVersion === "v2" &&
+            startMultipartResponse.multipartToken === undefined
+        ) {
+            throw new Error(
+                "Failed to start the multipart upload process, multipartToken does not exist in the response"
+            );
+        }
+
+        if (
+            this.opts.blobVersion === "v1" &&
+            (startMultipartResponse.uploadPartUrl === undefined ||
+                startMultipartResponse.completeUrl === undefined)
+        ) {
+            throw new Error(
+                `Failed to start the multipart upload process, required props (${[
+                    "uploadPartUrl",
+                    "completeUrl"
+                ].filter(
+                    p => !(p in startMultipartResponse)
+                )}) does not exist in the response`
```

Comment:
Looks good but is it really necessary? I mean these error strings are usually logged and handed over to the developers to investigate for bugs and issues. So does this make any difference to you if you get "The response did not contain x,y,z and we cannot continue, therefore we abort" or "Missing x,y,z; aborting upload"? The smaller the logs the better. E.g. same as above "Multipart upload v1 failed, uploadPartUrl=[true|false],  completeUrl=[true|false]"